src/cose/tags.rs

```diff
@@ -0,0 +1,208 @@
+//! Support for cbor tags
+use core::fmt;
+use core::marker::PhantomData;
+use serde::de::{
+    Deserialize, Deserializer, EnumAccess, IntoDeserializer, MapAccess, SeqAccess, Visitor,
+};
+use serde::forward_to_deserialize_any;
+use serde::ser::{Serialize, Serializer};
+
+/// signals that a newtype is from a CBOR tag
+pub(crate) const CBOR_NEWTYPE_NAME: &str = "\0cbor_tag";
+
+/// A value that is optionally tagged with a cbor tag
+///
+/// this only serves as an intermediate helper for tag serialization or deserialization
+pub struct Tagged<T> {
```

Comment:
I think you are re-implementing [`ciborium::tags::Captured`](https://docs.rs/ciborium/0.2.2/ciborium/tag/struct.Captured.html) here.

Reply:
will read more, but this existed before, I just moved it.